recursive_art.py

```diff
@@ -20,8 +25,31 @@ def build_random_function(min_depth, max_depth):
         (See the assignment writ-eup for details on the representation of
         these functions)
     """
-    # TODO: implement this
-    pass
+    random_list = ["prod", "avg", "cos_pi", "sin_pi"]
+    random_func = random.choice(random_list)
+    if max_depth > 0:
+        if min_depth > 0:
+            if random_func == "prod" or random_func == "avg":
+                return [random_func, build_random_function(min_depth-1, max_depth-1), build_random_function(min_depth-1, max_depth-1)]
+            else:
+                return [random_func, build_random_function(min_depth-1, max_depth-1)]
+        else:
+            random_list += ["x", "y"]
+            random_func = random.choice(random_list)
+            if random_func == "prod" or random_func == "avg":
+                return [random_func, build_random_function(min_depth-1, max_depth-1), build_random_function(min_depth-1, max_depth-1)]
+            elif random_func == "cos_pi" or random_func == "sin_pi":
+                return [random_func, build_random_function(min_depth-1, max_depth-1)]
+            else:
+                if random.randint(0,1) == 1:
+                    return "y"
+                else:
+                    return "x"
+    else:
+        if random.randint(0,1) == 1:
```

Comment:
You could use `return random.choice(["x", "y"])` here instead of another conditional.